blog-post/blog-post.md

```diff
@@ -0,0 +1,634 @@
+# Introducing Core Solidity
+
+Solidity is the most widely used smart contract language. It is robust, trustworthy, and today
+secures hundreds of billions of dollars of value. We are proud of this success, and its track
+record of secure code generation. Users of Solidity will however be keenly aware of some of its
+limitations. The type system often lacks the expressiveness to produce reusable library code or
+enforce core safety properties. The language has very limited capabilities around compile time
+evaluation. Many features are implemented in an inconsistent manner, or do not always work as
+expected.
+
+Fixing these limitations within the current implementation has proven difficult. Upgrades must be
+made in a somewhat ad-hoc manner, and each new feature addition makes reasoning about the
+correctness of subsequent features more difficult. We did not feel confident that we would be able
+to safely extend the language in this way to add the kind of features that our users were asking
+for, and that we feel are required to keep up with the ever increasing scale of systems that are
+being developed in Solidity.
+
+Core Solidity is our solution. It is a rebuild of the Solidity type system and compiler front/middle
+end that will:
+
+- introduce powerful new features (generics, traits, algebraic datatypes, pattern matching, comptime)
```

Comment:
```suggestion
- introduce powerful new features (generics, traits, algebraic datatypes, pattern matching, compile-time constant evaluation)
```

Unless you want to make "comptime" our official terminology (which I would not be against - I like things having unambiguous names). But AFAIK right now it's just a Zig term and not something used more widely.